tst/EnergyPlus/unit/ExtendedHI.unit.cc

```diff
@@ -0,0 +1,628 @@
+                                                  {219.9916, 219.9958, 220.0},
+                                                  {229.974, 229.987, 230.0},
+                                                  {239.9254, 239.9627, 240.0},
+                                                  {249.7677, 249.8837, 250.0},
+                                                  {259.3736, 259.6864, 260.0},
+                                                  {268.5454, 269.2746, 270.0},
+                                                  {277.2234, 278.6369, 280.0},
+                                                  {285.7511, 288.2814, 290.7861},
+                                                  {297.5738, 300.2923, 305.3947},
+                                                  {305.555, 318.6226, 359.9063},
+                                                  {313.0299, 359.0539, 407.5345},
+                                                  {320.5089, 398.576, 464.9949},
+                                                  {328.0358, 445.8599, 530.5525},
+                                                  {333.2806, 500.0422, 601.9518},
+                                                  {343.6313, 559.664, 677.2462},
+                                                  {354.1826, 623.196, 755.0833}};
+
+    std::vector<Real64> T_values = {200, 210, 220, 230, 240, 250, 260, 270, 280, 290, 300, 310, 320, 330, 340, 350, 360, 370};
+    std::vector<Real64> RH_values = {0, 0.5, 1};
+
+    Real64 tol = 1e-4;
+
+    for (size_t i = 0; i < T_values.size(); ++i) {
+        for (size_t j = 0; j < RH_values.size(); ++j) {
+            Real64 HI = HI_values[i][j];
+            EXPECT_NEAR(ExtendedHI::heatindex(*state, T_values[i], RH_values[j]), HI, tol);
+        }
+    }
+}
+
+// helper: calculating HI the old way
+Real64 calcHI(Real64 ZoneTF, Real64 ZoneRH)
+{
+    Real64 HI;
+    Real64 constexpr c1 = -42.379;
+    Real64 constexpr c2 = 2.04901523;
+    Real64 constexpr c3 = 10.14333127;
+    Real64 constexpr c4 = -.22475541;
+    Real64 constexpr c5 = -.00683783;
+    Real64 constexpr c6 = -.05481717;
+    Real64 constexpr c7 = .00122874;
+    Real64 constexpr c8 = .00085282;
+    Real64 constexpr c9 = -.00000199;
+    if (ZoneTF < 80) {
+        HI = 0.5 * (ZoneTF + 61.0 + (ZoneTF - 68.0) * 1.2 + (ZoneRH * 0.094));
+    } else {
+        HI = c1 + c2 * ZoneTF + c3 * ZoneRH + c4 * ZoneTF * ZoneRH + c5 * ZoneTF * ZoneTF + c6 * ZoneRH * ZoneRH + c7 * ZoneTF * ZoneTF * ZoneRH +
+             c8 * ZoneTF * ZoneRH * ZoneRH + c9 * ZoneTF * ZoneTF * ZoneRH * ZoneRH;
+        if (ZoneRH < 13 && ZoneTF < 112) {
+            HI -= (13 - ZoneRH) / 4 * std::sqrt((17 - abs(ZoneTF - 95)) / 17);
+        } else if (ZoneRH > 85 && ZoneTF < 87) {
+            HI += (ZoneRH - 85) / 10 * (87 - ZoneTF) / 5;
+        }
+    }
+    HI = (HI - 32.0) * (5.0 / 9.0);
+    return HI;
+}
+
+constexpr Real64 Fahrenheit2Celsius(Real64 F)
```

Comment:
I think there must be a function inside E+ code itself rather than another copy of it.

Reply:
I see. There's a TempIPtoSI function.